### ISSUE TITLE
Препроцессинг

### ISSUE THREAD

**Issue by iggisv9t** [ods-ai-ml4sg/proj_news_viz#125]
Нужно заменить в пайплайне скрипт препроцессинга и убедиться, что всё работает.
Скрипт есть готовый https://github.com/ods-ai-ml4sg/proj_news_viz/blob/master/nlp/news_viz/data/preprocessing_tools.py

По шагам:
1. Перенести скрипт указанный по ссылке на место такого же в папке pipeline
2. Убедиться, что оно не валится на импорте
3. Убедиться, что оно не валится на запуске
Как запустить: [см. инструкцию]( https://github.com/ods-ai-ml4sg/proj_news_viz/wiki/Pipeline-инструкция-по-применению). Нам нужен только `PreprocessorTask` поэтому заменяем `TopicPredictorTask` на `PreprocessorTask` в команде из инструкции.
4. Проверить, что результат корректный.
5. Закоммитить и сделать PR.

Всё правильно, если
1. Выполняется до конца без ошибок
2. Для каждого входного файла получается выходной файл, соответствующий [инструкции]( https://github.com/ods-ai-ml4sg/proj_news_viz/wiki/Pipeline-инструкция-по-применению)

**Comment by vtrokhymenko** [comment]
> Перенести скрипт указанный по ссылке на место такого же в папке pipeline

а зачем? это с пайплайна должны к нему обращаться

**Comment by iggisv9t** [comment]
> а зачем? это с пайплайна должны к нему обращаться  

Нам нужны разумные умолчания и чтобы пользователю не надо было знать всю историю проекта, чтобы найти подходящий файл. Если каждому надо будет делать каждый раз одно и то же действие: в данном случае взять тот файл и положить рядом, или поменять импорт на нужный путь, то не разумнее ли сделать это за пользователей один раз?
В нашеё структуре каталогов никто не сможет разобраться. Только три человека ей пользовались. Ты, потому что ты её придумал, я, потому что у меня не было вариантов лучше, и @Avenon и то он сначала просто всё складывал в experiments, а потом получилось, что часть лежит там, а часть в корне. При этом собственно продукта в nlp нет.

> а что такое «корректный» то?  

Хорошее замечание. Попробую описать, но боюсь не учесть всех нюансов.
1. Стоп слова удалены
2. Пунктуация удалена
3. Всё приведено к нижнему регистру (кстати, почему в скрипте этого нет?)
4. Все слова приведены в начальную форму
5. Текст в остальном ничего не потерял

**Comment by vtrokhymenko** [comment]
ну вот, тогда нужно переделать нормально пайплайн, а дублировать код вот точно не стоит
мы не полностью применили мой пайплайн, вот и получаем. изначально говорил чтобы то, что сейчас в `nlp` должно быть в самом корне, а ты решил иначе (без предъяв если что)

**Comment by vtrokhymenko** [comment]
> кстати, почему в скрипте этого нет?

наш лемматайзер автоматом приводит все к нижнему регистру

но замечание хорошее, лучше вынести в `return` это дело

**Comment by vtrokhymenko** [comment]
> Попробую описать, но боюсь не учесть всех нюансов.

по-хорошему, нужно тесты набить

**Comment by iggisv9t** [comment]
> а ты решил иначе (без предъяв если что)  

Да, я помню. Я потому что не мог въехать как оно будет работать и хотел понять но не перекраивать всё. Тем более у нас в активной работе было только nlp. Нужно снова всё переделывать, чтобы было норм. Но это уже не для этого issue диалог, давай в канале проекта, или отдельный ишью.

**Comment by buriy** [comment]
ох, не приводите слова к нижнему регистру -- потеряете порядка 10% качества на NER потом (95% -> 85%). Или просто предусмотрите возможность для NER делать без подобной нормализации, чтобы нормализация только для topic modeling была.

**Comment by vtrokhymenko** [comment]
это само собой, тк текущий препроцессинг не подходит для задачи ner, нужно отдельный писать

**Comment by buriy** [comment]
А у вас где-нибудь визуальная схема имеющихся пайплайнов есть?
Ну типа, 
[collector1], [collector2] -> [DB] -> [preprocessing] -> [TM] -> [visualizer]
что-то такое, но со ссылкой на конкретные .py файлы.

**Comment by vtrokhymenko** [comment]
в вики частично есть, но не так, как ты написал

**Comment by iggisv9t** [comment]
В самом ишью есть ссылка. В ридми в начале ссылка тоже. Там полное описание. Без визуальной схемы, но всё прямолинейно [preprocessing] -> [classifier] -> [TM].
https://github.com/ods-ai-ml4sg/proj_news_viz/wiki/Pipeline-инструкция-по-применению
  
DB пока не будет, потому что некому админить, а прикрутить её сверху на полностью рабочий пайплайн будет уже не так сложно. Визуализатор слишком сырой сейчас, и он не часть пайплайна, а клиент, скорее.
